threads/threads.tex

```diff
@@ -209,9 +209,8 @@ \section{Race Conditions}
 
 \begin{lstlisting}[language=C]
 void *thread_main(void *p) {
-  int x = *p;
-  x += x;
-  *p = x;
+  int *x = (int*) p;
+  *x = *x + *x;
```

Comment:
Reading this section in context, this is slightly different from the following assembly (in this section) assuming this code is translated directly w/ no optimizations. Technically every de-reference should be a load. I think it's probably better to change this to:

```c
int *p_int = (int *)p
int x = *p_int;
x += x;
*p_int = x;
```

on godbolt.org using gcc 9.3, this yeilds the following asm:
```asm
push    rbp
mov     rbp, rsp
mov     QWORD PTR [rbp-24], rdi
mov     rax, QWORD PTR [rbp-24]
mov     QWORD PTR [rbp-8], rax
mov     rax, QWORD PTR [rbp-8]
mov     eax, DWORD PTR [rax]
mov     DWORD PTR [rbp-12], eax
sal     DWORD PTR [rbp-12]
mov     rax, QWORD PTR [rbp-8]
mov     edx, DWORD PTR [rbp-12]
mov     DWORD PTR [rax], edx
nop
pop     rbp
ret
```
which is similar to the coursebook's content. I'd recommend not changing the coursebook's assembly lines though because those seem easier to understand. (e.g. explicitly adding instead of shifting to multiply by 2).

(For comparision here is the assembly for your change)
```asm
push    rbp
mov     rbp, rsp
mov     QWORD PTR [rbp-24], rdi
mov     rax, QWORD PTR [rbp-24]
mov     QWORD PTR [rbp-8], rax
mov     rax, QWORD PTR [rbp-8]
mov     eax, DWORD PTR [rax]
lea     edx, [rax+rax]
mov     rax, QWORD PTR [rbp-8]
mov     DWORD PTR [rax], edx
nop
pop     rbp
ret
```

Reply:
I did not consider the effect my change would have on the assembly. To be honest I can't read assembly that well so I am not confident enough to change it or even read it. Are you then suggesting that we keep the code as is? I'm personally fine with that. This is such a small change anyways that I think most people will be fine without.

Reply:
no the motivation for the change makes sense, just add casts to `(int*)` instead of changing the behavior. The assembly isn't that different, but it would be better if the assembly seemed to match exactly what's in the code.

Reply:
Okay I changed it and committed on this branch. Let me know.